packages/komencikit/src/kit.ts

```diff
@@ -0,0 +1,424 @@
+    const requestMetaTx = wallet.executeMetaTransaction(requestTx.txo, requestTxSig)
+
+    const resp = await this.client.exec(
+      requestSubsidisedAttestations({
+        identifier,
+        attestationsRequested,
+        walletAddress,
+        transactions: {
+          request: toRawTransaction(requestMetaTx.txo),
+          approve: toRawTransaction(approveMetaTx.txo),
+        },
+      })
+    )
+
+    if (!resp.ok) {
+      return resp
+    }
+
+    const txHash = resp.result.txHash
+    return this.waitForReceipt(txHash)
+  }
+
+  /**
+   * selectIssuers: wraps the `submitMetaTransaction` action in order
+   * to execute Attestations.selectIssuers(identifier)
+   *
+   * @param identifier - the phone number identifier
+   * @param walletAddress - MetaTransactionWallet address requesting attestations
+   */
+  public async selectIssuers(
+    identifier: string,
+    walletAddress: string
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const attestations = await this.contractKit.contracts.getAttestations()
+    await attestations.waitForSelectingIssuers(identifier, walletAddress)
+    return this.submitMetaTransaction(walletAddress, attestations.selectIssuers(identifier))
+  }
+
+  /**
+   * completeAttestation: wraps the `submitMetaTransaction` action in order
+   * to execute Attestations.complete(identifier, walletAddress, issuer, code)
+   *
+   * @param identifier - the phone number identifier
+   * @param walletAddress - MetaTransactionWallet address requesting attestations
+   * @param issuer - the issuer ID
+   * @param code - the code
+   */
+  public async completeAttestation(
+    identifier: string,
+    walletAddress: string,
+    issuer: Address,
+    code: string
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const attestations = await this.contractKit.contracts.getAttestations()
+    return this.submitMetaTransaction(
+      walletAddress,
+      await attestations.complete(identifier, walletAddress, issuer, code)
+    )
+  }
+
+  /**
+   * setAccount: wraps the `submitMetaTransaction` action in order
+   * to execute Account.setAccount(...)
+   *
+   * @param name A string to set as the name of the account
+   * @param dataEncryptionKey secp256k1 public key for data encryption. Preferably compressed.
+   * @param walletAddress The wallet address to set for the account
+   * @param proofOfPossession Signature from the wallet address key over the sender's address
+   */
+  public async setAccount(
+    name: string,
+    dataEncryptionKey: string,
+    walletAddress: Address,
+    proofOfPossession: Signature | null = null
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const accounts = await this.contractKit.contracts.getAccounts()
+    return this.submitMetaTransaction(
+      walletAddress,
+      accounts.setAccount(name, dataEncryptionKey, walletAddress, proofOfPossession).txo
```

Comment:
Don't you need to `await` this method call like you're doing elsewhere?

Reply:
It depends on the implementation in the wrapper - these calls result in a `CeloTransactionObject` but some are async, some are not.